source/credentials_provider_ecs.c

```diff
@@ -629,3 +628,93 @@ struct aws_credentials_provider *aws_credentials_provider_new_ecs(
 
     return NULL;
 }
+
+struct aws_credentials_provider *aws_credentials_provider_new_ecs_from_environment(
+    struct aws_allocator *allocator,
+    const struct aws_credentials_provider_ecs_environment_options *options) {
+
+    struct aws_credentials_provider_ecs_options explicit_options = {
+        .shutdown_options = options->shutdown_options,
+        .bootstrap = options->bootstrap,
+        .function_table = options->function_table,
+    };
+
+    struct aws_string *relative_uri_str = NULL;
+    struct aws_string *full_uri_str = NULL;
+    struct aws_uri full_uri;
+    AWS_ZERO_STRUCT(full_uri);
+    struct aws_credentials_provider *provider = NULL;
+
+    /* Read the auth token variables */
+    struct aws_string *ecs_env_token_file_path = NULL;
+    struct aws_string *ecs_env_token = NULL;
+    if (aws_get_environment_value(allocator, s_ecs_creds_env_token_file, &ecs_env_token_file_path) != AWS_OP_SUCCESS ||
+        aws_get_environment_value(allocator, s_ecs_creds_env_token, &ecs_env_token) != AWS_OP_SUCCESS) {
+        goto cleanup;
+    }
+    if (ecs_env_token_file_path != NULL) {
+        explicit_options.auth_token_file_path = aws_byte_cursor_from_string(ecs_env_token_file_path);
+    }
+    if (ecs_env_token != NULL) {
+        explicit_options.auth_token = aws_byte_cursor_from_string(ecs_env_token);
+    }
+
+    if (aws_get_environment_value(allocator, s_ecs_creds_env_relative_uri, &relative_uri_str) == AWS_OP_SUCCESS &&
```

Comment:
trivial: if you chose not to check this error above, then also don't check it here. Maybe just put all 4 read-env-var calls in a row